src/ResourceManager/Sql/Commands.Sql/Security/Cmdlet/SqlCmdletBase.cs

```diff
@@ -37,47 +41,57 @@ public abstract class SqlDatabaseSecurityCmdletBase : SqlDatabaseCmdletBase
+        public A ModelAdapter { get;  internal set; }
 
         /// <summary>
-        /// Provides the auditing policy that this cmdlet operates on
+        /// Provides the model element that this cmdlet operates on
         /// </summary>
-        /// <returns>An auditingPolicy object</returns>
-        protected abstract AuditingPolicy GetPolicy();
-         
+        /// <returns>A model object</returns>
+        protected abstract M GetModel();
+
         /// <summary>
-        /// Updates the given policy with the cmdlet specific operation 
+        /// This method is responsible to call the right API in the communication layer that will eventually send the information in the 
+        /// object to the REST endpoint
         /// </summary>
-        /// <param name="policy">An AuditingPolicy object representing the policy of the current resource</param>
-        protected virtual void UpdatePolicy(AuditingPolicy policy) { }
-        
+        /// <param name="policy">The model object with the data to be sent to the REST endpoints</param>
+        protected virtual void SendModel(M model) { }
+         
         /// <summary>
-        /// This method is responsible to call the right API that eventually send the information found in the given AuditingPolicy
-        /// object to the backend
+        /// Updates the given model element with the cmdlet specific operation 
         /// </summary>
-        /// <param name="policy">The AuditingPolicy object with the data to be sent to the backend</param>
-        protected virtual void SendPolicy(AuditingPolicy policy) { }
+        /// <param name="model">A model object</param>
+        protected virtual M UpdateModel(M model) { return model; }
 
         /// <summary>
-        /// Returns true if the AuditingPolicy that was developed by this cmdlet should be written out
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
         /// </summary>
-        /// <returns>True if the AuditingPolicy should be written out, False otherwise</returns>
-        protected virtual bool writeResult() { return true; }
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected virtual bool WriteResult() { return true; }
+
+        protected abstract A InitModelAdapter (AzureSubscription subscription);
+
         
         /// <summary>
         /// Executes the cmdlet
         /// </summary>
         public override void ExecuteCmdlet()
         {
-            PolicyHandler = new SqlClient(Profile, Profile.Context.Subscription);
-            AuditingPolicy policy = this.GetPolicy();
-            this.UpdatePolicy(policy);
-            this.SendPolicy(policy);
-            if (writeResult()) this.WriteObject(policy);
+            ModelAdapter = InitModelAdapter(Profile.Context.Subscription); 
+            M model = this.GetModel();
+            M updatedModel = this.UpdateModel(model);
```

Comment:
All the Get-XXX do not use the UpdateModel. I think that since these two have a very different semantics, it would be better to keep them separated.